system-addon/lib/TopSitesFeed.jsm

```diff
@@ -190,6 +197,33 @@ this.TopSitesFeed = class TopSitesFeed {
     }
   }
 
+  /**
+   * Set a custom screenshot for a topsite.
+   * @param url {string} The topsite URL to update with the new screenshot
+   * @param screenshotURL {string} The URL to use for the new screenshot
+   */
+  async setCustomTopsiteScreenshot({url, customScreenshotURL}) {
+    const pinnedLinks = await this.pinnedCache.request();
+    const frecentLinks = await this.frecentCache.request();
+    // Get the LinksCache entry that represents the topsite to update.
+    const link = pinnedLinks.concat(frecentLinks).find(topsite =>
+      topsite && topsite.url === url);
+
+    if (link) {
+      // If previous request failed, we need to clear any cached result
+      link.screenshotPreview = undefined;
+      Screenshots.maybeCacheScreenshot(link, customScreenshotURL, "screenshotPreview",
```

Comment:
Why do we want to cache the preview image?

Reply:
Replaced with `getScreenshotForURL`.